.github/workflows/codeql-analysis.yml

```diff
@@ -5,10 +5,14 @@ on:
     branches:
       - main
       - 'rc/*'
+    paths:
+      - 'csharp/**'
```

Comment:
To give an alert diff on the PR we need to analyze every commit on the base branches to make sure we have results there to compare against, so I'd suggest reverting this part. (https://docs.github.com/en/enterprise-server@2.22/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning#avoiding-unnecessary-scans-of-pull-requests)

Reply:
Thanks for raising this. I created a PR with the fix: https://github.com/github/codeql/pull/5299